tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/DotnetTest/DotnetCommon.cs

```diff
@@ -40,10 +42,46 @@ internal static bool IsDataCollectorDomain
                     return value;
                 }
 
+                // If the AppDomainName contains "DataCollector" we are in the data collector domain
                 return (_isDataCollectorDomainCache = DomainMetadata.Instance.AppDomainName.ToLowerInvariant().Contains("datacollector")).Value;
             }
         }
 
+        internal static bool IsMsBuildTask
+        {
+            get
+            {
+                if (_isMsBuildTaskCache is { } value)
+                {
+                    return value;
+                }
+
+                // Let's try to detect if we are in the MSBuild task scenario
+                // the process name is not guaranteed so we need to check the stack trace
+                // to see if we are getting called from the Coverlet.MSbuild.Tasks namespace
+                // because we don't need any symbols this should be fast enough.
+                if (new StackTrace(3, false).GetFrames() is { } frames)
+                {
+                    foreach (var stackFrame in frames)
+                    {
+                        if (stackFrame is null)
+                        {
+                            continue;
+                        }
+
+                        if (stackFrame.GetMethod()?.DeclaringType?.FullName?.Contains("Coverlet.MSbuild.Tasks") == true)
```

Comment:
nit: `Contains` or `StartsWith`?

Reply:
done in [5bcd7ae](https://github.com/DataDog/dd-trace-dotnet/pull/6284/commits/5bcd7aef97145bdad1950e684d0738196e282ef4)